lib/membrane_live_web/channels/event_channel.ex

```diff
@@ -248,4 +209,88 @@ defmodule MembraneLiveWeb.EventChannel do
         {:reply, {:ok, GenServer.call(socket.assigns.event_pid, :is_playlist_playable)}, socket}
     end
   end
+
+  defp webinar_exists(id) do
+    case Ecto.UUID.cast(id) do
+      {:ok, id} ->
+        {:ok, Repo.exists?(from(w in Webinar, where: w.uuid == ^id))}
+
+      _error ->
+        {:error, "id is not binary_id"}
+    end
+  end
+
+  defp create_event_stream(event_id, socket) do
+    case :global.whereis_name(event_id) do
+      :undefined -> Event.start(event_id, name: {:global, event_id})
+      pid -> {:ok, pid}
+    end
+    |> case do
+      {:ok, event_pid} ->
+        {:ok, Phoenix.Socket.assign(socket, %{event_id: event_id, event_pid: event_pid})}
+
+      {:error, {:already_started, event_pid}} ->
+        {:ok, Phoenix.Socket.assign(socket, %{event_id: event_id, event_pid: event_pid})}
+
+      {:error, reason} ->
+        Logger.error("""
+        Failed to start room.
+        Room: #{inspect(event_id)}
+        Reason: #{inspect(reason)}
+        """)
+
+        {:error, %{reason: "failed to start event"}}
+    end
+  end
+
+  defp join_event_stream(socket) do
+    peer_id = "#{UUID.uuid4()}"
+    # TODO handle crash of room?
+    Process.monitor(socket.assigns.event_pid)
+    send(socket.assigns.event_pid, {:add_peer_channel, self(), peer_id})
+
+    {:ok, Phoenix.Socket.assign(socket, %{peer_id: peer_id})}
+  end
+
+  defp check_if_presenter(email, reloaded, id),
+    do: check_if_exist_in_ets(:presenters, email, reloaded, id)
+
+  defp check_if_request_presenting(email, reloaded, id),
+    do: check_if_exist_in_ets(:presenting_requests, email, reloaded, id)
+
+  defp check_if_exist_in_ets(ets_key, email, reloaded, id) do
+    [{_key, presenters}] = :ets.lookup(ets_key, id)
+    in_ets = MapSet.member?(presenters, email)
+
+    case {reloaded, in_ets} do
+      {true, _in_ets} ->
+        {:ok, in_ets}
+
+      {false, true} ->
+        remove_from_presenters(email, id)
+        {:ok, false}
+
+      {false, false} ->
+        {:ok, false}
+    end
+  end
+
+  defp remove_from_presenters(email, id), do: remove_from_list_in_ets(:presenters, email, id)
+  defp add_to_presenters(email, id), do: add_to_list_in_ets(:presenters, email, id)
+
+  defp remove_from_presenting_requests(email, id),
+    do: remove_from_list_in_ets(:presenting_requests, email, id)
+
+  defp add_to_presenting_request(email, id),
+    do: add_to_list_in_ets(:presenting_requests, email, id)
+
+  defp remove_from_list_in_ets(ets_key, email, id) do
+    [{_key, presenters}] = :ets.lookup(ets_key, id)
+    :ets.insert(ets_key, {id, MapSet.delete(presenters, email)})
+  end
+
+  defp add_to_list_in_ets(ets_key, email, id) do
+    [{_key, presenters}] = :ets.lookup(ets_key, id)
+    :ets.insert(ets_key, {id, MapSet.put(presenters, email)})
```

Comment:
`event_channel.exs` has become quite big with almost 300 lines. Wouldn't it be nice to delgate some functions to i.e. `event_channel_helpers.ex`?

Reply:
I looked again through this file and the only functions we could move to the separate module are the current private functions. As the logic of these functions isn't shared with another module I don't see much benefit from moving it to a separate module.